### DISCUSSION TITLE
Invalid Characters in Variable

### DISCUSSION THREAD

**Discussion by mrussell1958** [mpaperno/MSFSTouchPortalPlugin#56] [Q&A] [answered]
This ay be a dumb question but I will ask it anyway. I am trying to get the engine readouts for N1, EGT etc fromthe FWB A320. I was using the MSFS States Editor V3 to check the response. I requested A32NX_ENGINE_N1:1 from the drop down, but when I press the button I get a PLugin Error .. Variable Request Validation error "Variable Name 'A32NX_ENGINE_N!:!' contains invalid characters........ I am not sure how to fix this as I cant change the characters, and even if I could what would I change them to. Any help for a novewould be appreciated.

**Comment by mpaperno** [comment]
Hi,

Seems like a reasonable question to me... :)

I'm not sure why it would show that error unless the variable name actually had `!`s instead of `1`s.  But could be some bug I suppose...

Could you paste a screenshot of the TP button setup you're using to make this request?  I'm not sure which action you're actually using or which drop down you're selecting from.

I'm also wondering if `A32NX_ENGINE_N1` really has an index value (the `:1` part) since it looks like an L var, not a Sim Var ("A" type).

Cheers,
-Max

**Reply by mrussell1958** [reply]
I have been trying to fix the overhead pannel display which currently showing the text in green thats in the OnEvent Update Button Text. Currently its set to 

${value:MSFSTouchPortalPlugin.Engine.State.GsA32nxEngine1N1Value}

${value:MSFSTouchPortalPlugin.Engine.State.GsA32nxEngine1EGT} 

So I was trying to find what the test page would return for these variables. So when I go to set up the "Aeroplane Local Variable button" on the test page I have obviously chosen the wrong variable to request.

Now having spent more time on it there doesn't appear to be a variable simply for N1 or the others. Perhaps FBW have changed this. Thanks for the very prompt answer, I am new to this with little experience. Up to now I concentrated on AAO's and RPN, but I won't be beaten :-)

Once I get to the bottom I will update the comments on flightsim.to with my findings and fix (hopefully)

**Comment by Pyrolator** [comment]
See this Discussion to late and opend an Issue (#57).
Same Engine-Error for me plus not working Buttons in ECAM.
Sadly im a Novice about programming and so i cant fix it by myself :(

**Comment by Pyrolator** [comment]
Hello,
the Tipp from the TO is to use the 1.3.2 TPP.
That olderplugin sees also the values for the Engines.

Max, I hope that can be fixed in a future release ?

Greets
Roger

**Comment by mpaperno** [comment, marked as answer]
Belatedly updating/closing this. Fixed a couple versions ago in [v1.4.2](https://github.com/mpaperno/MSFSTouchPortalPlugin/releases/tag/1.4.2.0).
